### PR TITLE
Refactoring print

### PR DESCRIPTION
**Pull request by arbiocanpion**
## What?
Separate the prints and added a new class who interacts with the shell

## Why?
Becauses  the goor architecture its like that

### CODE REVIEW

File: lib/ui.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  attr_reader :input
+
+  def print_console(input)
+    puts input
+  end
+
+  def say(input = '')
+    say(input)
+  end
+end
```
Comment (dasolari) [inline lib/ui.rb:16]: This class should have been implemented as a Singleton like we discussed.
https://refactoring.guru/design-patterns/singleton/ruby/example

File: lib/observer/observable.rb [hunk 2]
```diff
@@ -1,9 +1,12 @@
...
+require_relative '../ui'
+
 # Class for creating an observable object following the observer desgin pattern
 class Observable
   def initialize
     @observers = []
+    @ui = Ui.new
   end
 
   def add_observer(observer)
```
Comment (dasolari) [inline lib/observer/observable.rb:12]: I think this class has to go, it follows the code smell Speculative Generality.

**Review by dasolari — CHANGES_REQUESTED** [review]
Observable and observer must be deleted, they are unused. I suggested a change with the implementation of the UI class.

File: lib/ui.rb (hunk 1 above)
Comment (dasolari) [reply lib/ui.rb:16]: Also, I suggest changing `# Class who administrate the prints` to `# Class that administrates ui inputs and prints`

**Review by Meaylwin — APPROVED** [review]

**Review by dasolari — APPROVED** [review]
👍

**Pull request merged**